data/kernels/blendop.cl

```diff
@@ -941,4 +1014,117 @@ blendop_set_mask (__write_only image2d_t mask, const int width, const int height
 }
 
 
+__kernel void
+blendop_display_channel (__read_only image2d_t in_a, __read_only image2d_t in_b, __read_only image2d_t mask, __write_only image2d_t out, const int width, const int height, 
+                         const int2 offs, const int mask_display)
+{
+  const int x = get_global_id(0);
+  const int y = get_global_id(1);
+
+  if(x >= width || y >= height) return;
+
+  float4 a = read_imagef(in_a, sampleri, (int2)(x, y) + offs);
+  float4 b = read_imagef(in_b, sampleri, (int2)(x, y));
+  float opacity = read_imagef(mask, sampleri, (int2)(x, y)).x;
+
+  float c;
+  float4 LCH;
+  float4 HSL;
+
+  dt_dev_pixelpipe_display_mask_t channel = (dt_dev_pixelpipe_display_mask_t)mask_display;
+  
+  switch(channel & DT_DEV_PIXELPIPE_DISPLAY_ANY)
```

Comment:
No issue. Neither with NVIDIA nor AMD. In fact size does not seem to be a determining factor when it comes to OpenCL compiler bugs. I have seen the biggest problems with kernels of just a few lines.